### PR TITLE
Java execute batch - New API Integration

### PR DESCRIPTION
**Pull request by harshit-gangal**

### CODE REVIEW

**Comment by harshit-gangal** [comment]
I will add E2E test.

**Comment by harshit-gangal** [comment]
* Java E2E test is added.
* API is fixed
* Proto3 conversion unit test added.

**Comment by sougou** [comment]
@dumbunny The Go parts LGTM.

File: java/client/src/main/java/com/youtube/vitess/client/Proto.java [hunk 1]
```diff
@@ -217,6 +219,14 @@ public static BoundKeyspaceIdQuery bindKeyspaceIdQuery(String keyspace,
     return builder.build();
   }
 
+  public static List<CursorWithError> fromQueryResponsestoCursorList(List<Query.ResultWithError> resultWithErrorList) {
```
Comment (dumbunny) [inline java/client/src/main/java/com/youtube/vitess/client/Proto.java]: Please capitalize "to" here and in files that use this method.

File: java/client/src/main/java/com/youtube/vitess/client/VTGateConn.java [hunk 2]
```diff
@@ -202,7 +205,46 @@ public VTGateConn(RpcClient client, String keyspace) {
...
+    }
+
+    public SQLFuture<List<CursorWithError>> executeBatch(Context ctx, List<String> queryList,
+        @Nullable List<Map<String, ?>> bindVarsList, TabletType tabletType,
+        boolean asTransaction) throws SQLException {
+        List<Query.BoundQuery> queries = new ArrayList<>();
+
+        if (null != bindVarsList && bindVarsList.size() != queryList.size()) {
+            throw new SQLDataException(
+                "Number of SQL Query list does not match the bind variables list");
```
Comment (dumbunny) [inline java/client/src/main/java/com/youtube/vitess/client/VTGateConn.java]: s/Number/Size/

File: java/client/src/main/java/com/youtube/vitess/client/VTGateTx.java [hunk 3]
```diff
@@ -198,6 +202,40 @@
...
   }
 
+    public SQLFuture<List<CursorWithError>> executeBatch(Context ctx, List<String> queryList,
+        @Nullable List<Map<String, ?>> bindVarsList, TabletType tabletType)
+        throws SQLException {
+        List<Query.BoundQuery> queries = new ArrayList<>();
+
+        if (null != bindVarsList && bindVarsList.size() != queryList.size()) {
+            throw new SQLDataException(
+                "Number of SQL Query list does not match the bind variables list");
```
Comment (dumbunny) [inline java/client/src/main/java/com/youtube/vitess/client/VTGateTx.java]: s/Number/Size/

**Pull request merged**